__tests__/app.test.js

```diff
@@ -335,4 +335,64 @@ describe('POST /api/reviews/:review_id/comments', () =>
             .expect(500);
         });
     });
-});
\ No newline at end of file
+});
+
+
+
+
+
+
+
+describe('GET /api/users', () =>
+{
+    test('server sends response with 200 status code and an array of user objects', () =>
+    {
+        return request(app)
+        .get('/api/users')
+        .expect(200)
+        .then((response) =>
+        {
+            const { users } = response.body;
+            expect(users).not.toBe(undefined);
+
+            expect(users.length).toBeGreaterThan(0);
+
+            for(let user of users)
+                expect(typeof user).toBe('object');
+        });
+    });
+
+    test('to check each user object has username,name & avatar_url properties', () =>
+    {
+        return request(app)
+        .get('/api/users')
+        .expect(200)
+        .then((response) =>
+        {
+            const { users } = response.body;
+
+            expect(users.length).toBeGreaterThan(0);
+
+            for(let user of users)
+            {
+                expect(user).toHaveProperty('username', expect.any(String));
+                expect(user).toHaveProperty('name', expect.any(String));
+                expect(user).toHaveProperty('avatar_url', expect.any(String));
+            }    
+        });
+    });
+
+    describe('Errors', () =>
```

Comment:
The 404 for incorrect path only needs to be tested once in your api, as it will catch all occurrences of mistyped paths.

Reply:
sure.